### PR TITLE
[Core/Mapping] move Mapper interface to Core

### PR DESCRIPTION
**Pull request by philbucher**
As discussed for a long time already, this PR makes a first step in integrating the mapper in the Core. This part is very consolidated for years already.
In a first step I am moving the baseclass/interface as well as the corresponding factory

The rest (i.e. the consolidated mappers) will be moved once I use the Kratos CSR matrix (which will make them independent from Trilinos in MPI)

The changes are done fully backward compatible!

FYI @KratosMultiphysics/cosimulation, @rubenzorrilla, @marcnunezc

### CODE REVIEW

File: applications/MappingApplication/MappingApplication.py [hunk 1]
```diff
@@ -24,6 +26,9 @@ def GetRegisteredMPIMapperNames(*args):
     from KratosMultiphysics.MappingApplication.MPIExtension import MPIMapperFactory
     return MPIMapperFactory.GetRegisteredMapperNames(*args)
 
+MapperFactory = CoreMapperFactory
+Mapper = CoreMapper
```
Comment (roigcarlo) [inline applications/MappingApplication/MappingApplication.py]: Why not just:

```Python
from KratosMultiphysics import Mapper
from KratosMultiphysics import MapperFactory
```

Seems like a double indirection as it is right now. Are the `CoreXXX` alias used or planed to be used anywhere?

File: applications/MappingApplication/mpi_extension/custom_python/mapping_mpi_extension.cpp [hunk 2]
```diff
@@ -21,15 +21,15 @@
...
+#include "factories/mapper_factory.h"
+#include "python/add_mapper_to_python.h"
 
 namespace Kratos {
 namespace Python {
 
 PYBIND11_MODULE(KratosMappingMPIExtension,m)
 {
-    AddMapperToPython<MPIMapperDefinitions::SparseSpaceType, MPIMapperDefinitions::DenseSpaceType>(m);
+    AddMappingToPython<MPIMapperDefinitions::SparseSpaceType, MPIMapperDefinitions::DenseSpaceType>(m);
```
Comment (roigcarlo) [inline applications/MappingApplication/mpi_extension/custom_python/mapping_mpi_extension.cpp:32]: Not this line in particular but, why is the mapper interface moved to the core and the mpi extension is still in the application? shouldn't we move the interface of the extension to `KratosMPICore` as well?

File: applications/MappingApplication/MappingApplication.py (hunk 1 above)
Comment (philbucher) [reply applications/MappingApplication/MappingApplication.py]: good point, not sure why I didn't do it in that way...

File: applications/MappingApplication/mpi_extension/custom_python/mapping_mpi_extension.cpp (hunk 2 above)
Comment (philbucher) [reply applications/MappingApplication/mpi_extension/custom_python/mapping_mpi_extension.cpp:32]: unfortunately the MPI extension still depends on Trilinos which is why I cannot move it to the Core
I am currently refactoring with the Kratos-CSR matrices, once I am done with that I will move it as you suggest

**Comment by pooyan-dadvand** [comment]
Good move! :+1:

**Review by roigcarlo — APPROVED** [review]
Great :+1:

**Pull request merged**